### PR TITLE
docs: make style guide fixes throughout

### PR DESCRIPTION
**Pull request by pierwill**
## Status

Ready for review

## Description of Changes

Mostly adding italics to the term "Journalist Interface"

## Checklist

- [x] Doc linting (`make docs-lint`) passed locally

### CODE REVIEW

File: docs/install.rst [hunk 1]
```diff
@@ -25,10 +25,10 @@ on network speed and computing power.
 
 .. _configure_securedrop:
 
-Localization of the source and journalist interfaces
-----------------------------------------------------
+Localization of the User Interfaces
```
Comment (dachary) [inline docs/install.rst]: We do not have **User Interfaces** anywhere else. Would it be a problem to use 

    *Source Interface* and *Journalist Interface* 

instead?

File: docs/threat_model/threat_model.rst [hunk 2]
```diff
@@ -337,14 +337,14 @@ What a compromise of the journalist's property can achieve
...
-   -  Access the Hidden Service values used by the Journalist Interface.
+   -  Access the Hidden Service values used by the *Journalist Interface*.
    -  Access SSH keys and passphrases for the *Application Server* and the
       *Monitor Server*.
    -  Access the journalist's personal GPG key.
 
 -  An attacker with journalist access to the *Journalist Interface* can:
 
-   -  Change the codenames associated with sources within the Interface.
+   -  Change the codenames associated with sources within the interface.
```
Comment (dachary) [inline docs/threat_model/threat_model.rst]: I don't think we should use a capitalize form here.

File: docs/install.rst (hunk 1 above)
Comment (pierwill) [reply docs/install.rst]: :heavy_check_mark:

File: docs/threat_model/threat_model.rst (hunk 2 above)
Comment (pierwill) [reply docs/threat_model/threat_model.rst]: I agree. The diff looks like I did make it lowercase, but I'll check again.

Comment (dachary) [reply docs/threat_model/threat_model.rst]: Oh, you are right! I read it the other way around, my bad.

**Pull request merged**